include/my_global.h

```diff
@@ -617,6 +617,9 @@ typedef SOCKET_SIZE_TYPE size_socket;
 #else
 #define STR_O_CLOEXEC ""
 #endif
+#ifndef O_TMPFILE
+#define O_TMPFILE       0
+#endif
 #ifndef SOCK_CLOEXEC
```

Comment:
No need to. If O_TMPFILE isn't supposed to be used outside of mf_tempfile.c, it'd be less confusing not to define it globally.